File: backend/pdf2image.py
import fitz
import sys

pdf_path = sys.argv[1]
page = sys.argv[2]
dpi = sys.argv[3]
output = sys.argv[4]

doc = fitz.open(pdf_path)
page = doc.load_page(int(page))
pix = page.get_pixmap(dpi = int(dpi))
pix.save(output)

File: backend/pdf2image.ts
import { exec, execSync } from 'node:child_process';
import { join } from 'node:path/posix';

export default async function pdf2image({
  pdf_path,
  page,
  dpi = 150,
  output = join(process.cwd(), 'tmp.png'),
}: {
  pdf_path: string;
  page: number;
  dpi?: number;
  output?: string;
}): Promise<string> {
  const py_command = `python3 backend/pdf2image.py ${pdf_path} ${page} ${dpi} ${output}`;
  execSync(py_command).toString();
  return output;
}
